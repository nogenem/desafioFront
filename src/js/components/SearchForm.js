import { sanitize } from "../utils";

const renderSearchForm = () => `
  <div id="searchContainer" class="row align-items-center justify-content-center">
    <form id="searchForm" class="col-sm-8 col-md-6 col-lg-6">
      <div class="form-group">
        <label for="userSearchInput">Usuário:</label>
        <input 
          id="userSearchInput" 
          type="search" 
          class="form-control" 
          placeholder="Usuário para busca..."
        >
      </div>
    </form>
  </div>
`;

export default {
  render: renderSearchForm,
  didMount: () => {
    $("#searchForm").submit(e => {
      e.preventDefault();
      const user = sanitize($("#searchForm input").val());

      location.hash = `#/user/${user}`;
    });
  }
};
