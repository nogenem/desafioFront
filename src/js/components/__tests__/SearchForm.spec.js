import SearchForm from "../SearchForm";

it("renders correctly", () => {
  const output = SearchForm.render();
  expect(output).toMatchSnapshot();
});
