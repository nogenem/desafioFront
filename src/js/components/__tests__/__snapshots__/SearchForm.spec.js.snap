// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"
  <div id=\\"searchContainer\\" class=\\"row align-items-center justify-content-center\\">
    <div class=\\"col-12\\">
      <div class=\\"row align-items-center justify-content-center text-center\\">
        <div class=\\"col-6\\">
          <img 
            src=\\"http://www.freepngimg.com/download/github/1-2-github-free-png-image.png\\" 
            alt=\\"Github logo\\" class=\\"img-fluid\\">
        </div>
      </div>
      <div class=\\"row align-items-center justify-content-center\\">
        <form id=\\"searchForm\\" class=\\"col-12 col-sm-12 col-md-8 col-lg-6 col-xl-6\\">
          <div class=\\"form-group input-group\\">
            <input 
              id=\\"userSearchInput\\" 
              type=\\"search\\" 
              class=\\"form-control\\" 
              placeholder=\\"UsuÃ¡rio para busca...\\"
            >
          </div>
        </form>
      </div>
    </div>
  </div>
"
`;
